expression/constant_fold.go

```diff
@@ -56,12 +56,8 @@ func ifFoldHandler(expr *ScalarFunction) (Expression, bool) {
 		}
 		return foldConstant(args[2])
 	}
-	var isDeferred, isDeferredConst bool
-	expr.GetArgs()[1], isDeferred = foldConstant(args[1])
-	isDeferredConst = isDeferredConst || isDeferred
-	expr.GetArgs()[2], isDeferred = foldConstant(args[2])
-	isDeferredConst = isDeferredConst || isDeferred
-	return expr, isDeferredConst
+	// if the condition is not const, which branch is unknown to run, so directly return.
+	return expr, false
```

Comment:
There probably are some constants in its arguments that can be folded. For example:
```sql
select if(a, 1+100, 2+100) from t;
```

Reply:
for control expression, it involves two steps: its all parameters (branches), itself.

1. When invocking `Enter`, each branch of control expressions fold contrants whenever possible, but not throw uncessary warnings/errors.
2. folding the control expression itself.

So the constants are folded with best. 
Your case is also folded.